pkg/cluster/internal/create/actions/init/init.go

```diff
@@ -0,0 +1,168 @@
+		"--ignore-preflight-errors=all",
+		// specify our generated config file
+		"--config=/kind/kubeadm.conf",
+		"--skip-token-print",
+		// increase verbosity for debugging
+		"--v=6",
+	)
+	lines, err := exec.CombinedOutputLines(cmd)
+	log.Debug(strings.Join(lines, "\n"))
+	if err != nil {
+		return errors.Wrap(err, "failed to init node with kubeadm")
+	}
+
+	// copies the kubeconfig files locally in order to make the cluster
+	// usable with kubectl.
+	// the kubeconfig file created by kubeadm internally to the node
+	// must be modified in order to use the random host port reserved
+	// for the API server and exposed by the node
+
+	// retrives the random host where the API server is exposed
+	// TODO(fabrizio pandini): when external load-balancer will be
+	//      implemented this should be modified accordingly
+	hostPort, err := node.Ports(kubeadm.APIServerPort)
+	if err != nil {
+		return errors.Wrap(err, "failed to get kubeconfig from node")
+	}
+
+	kubeConfigPath := ctx.ClusterContext.KubeConfigPath()
+	if err := node.WriteKubeConfig(kubeConfigPath, hostPort); err != nil {
+		return errors.Wrap(err, "failed to get kubeconfig from node")
+	}
+
+	// install the CNI network plugin
+	// TODO(bentheelder): this should possibly be a different action?
+	// TODO(bentheelder): support other overlay networks
+	// first probe for a pre-installed manifest
+	haveDefaultCNIManifest := true
+	if err := node.Command("test", "-f", "/kind/manifests/default-cni.yaml").Run(); err != nil {
+		haveDefaultCNIManifest = false
+	}
+	if haveDefaultCNIManifest {
+		// we found the default manifest, install that
+		// the images should already be loaded along with kubernetes
+		if err := node.Command(
+			"kubectl", "create", "--kubeconfig=/etc/kubernetes/admin.conf",
+			"-f", "/kind/manifests/default-cni.yaml",
+		).Run(); err != nil {
+			return errors.Wrap(err, "failed to apply overlay network")
+		}
+	} else {
+		// fallback to our old pattern of installing weave using their recommended method
+		if err := node.Command(
+			"/bin/sh", "-c",
+			`kubectl apply --kubeconfig=/etc/kubernetes/admin.conf -f "https://cloud.weave.works/k8s/net?k8s-version=$(kubectl version --kubeconfig=/etc/kubernetes/admin.conf | base64 | tr -d '\n')"`,
+		).Run(); err != nil {
+			return errors.Wrap(err, "failed to apply overlay network")
+		}
+	}
+
+	// if we are only provisioning one node, remove the master taint
+	// https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/#master-isolation
+	if len(allNodes) == 1 {
+		if err := node.Command(
+			"kubectl", "--kubeconfig=/etc/kubernetes/admin.conf",
+			"taint", "nodes", "--all", "node-role.kubernetes.io/master-",
+		).Run(); err != nil {
+			return errors.Wrap(err, "failed to remove master taint")
+		}
+	}
+
+	// add the default storage class
+	// TODO(bentheelder): this should possibly be a different action?
+	if err := addDefaultStorageClass(node); err != nil {
```

Comment:
+1

Reply:
also this storage class is basically bogus, we should ship #118 soon